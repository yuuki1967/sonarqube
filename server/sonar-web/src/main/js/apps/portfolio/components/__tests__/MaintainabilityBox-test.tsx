/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import MaintainabilityBox from '../MaintainabilityBox';

it('renders', () => {
  const measures = {
    sqale_rating: '3',
    last_change_on_maintainability_rating: '{"date":"2017-01-02T00:00:00.000Z","value":2}',
    maintainability_rating_effort: '{"rating":3,"projects":1}'
  };
  expect(shallow(<MaintainabilityBox component="foo" measures={measures} />)).toMatchSnapshot();
});
