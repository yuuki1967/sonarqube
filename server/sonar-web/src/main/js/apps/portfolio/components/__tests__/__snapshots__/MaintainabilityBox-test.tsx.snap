// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="portfolio-box portfolio-maintainability"
>
  <h2
    className="portfolio-box-title"
  >
    metric_domain.Maintainability
    <MeasuresButtonLink
      component="foo"
      metric="Maintainability"
    />
    <HistoryButtonLink
      component="foo"
      metric="sqale_rating"
    />
  </h2>
  <MainRating
    component="foo"
    metric="sqale_rating"
    value="3"
  />
  <RatingFreshness
    lastChange="{\\"date\\":\\"2017-01-02T00:00:00.000Z\\",\\"value\\":2}"
    rating="3"
  />
  <Effort
    component="foo"
    effort={
      Object {
        "projects": 1,
        "rating": 3,
      }
    }
    metricKey="sqale_rating"
  />
</div>
`;
